fix backup import process to validate structure and move files correctly

erhanurgun opened the pull request:
For details, see: https://github.com/vitodeploy/vito/issues/801

Some of the situations described here have been corrected, but not all of them...

erhanurgun committed:
fix backup import process to validate structure and move files correctly

erhanurgun committed:
The accidentally deleted comment lines have been restored.

erhanurgun committed:
use File::exists for path validation in VitoSettingController

erhanurgun committed:
refactor backup file handling to improve structure and add missing sources

erhanurgun committed:
enhance backup file validation to include additional MIME types

saeedvaziry commented:
@erhanurgun would you mind adding a test to cover it?

erhanurgun commented:
> @erhanurgun would you mind adding a test to cover it?

<img width="1270" height="539" alt="image" src="https://github.com/user-attachments/assets/031e6096-10f9-4447-be67-eb6ed3afb808" />

**Test coverage added!**

  I've created comprehensive tests for the backup import functionality in `VitoSettingsImportTest.php`. The tests cover all the changes made in this branch:

  **Tests Added:**
  - ✅ `test_import_backup_with_valid_zip_file` - Valid backup import
  - ✅ `test_import_backup_rejects_invalid_file_type` - File type validation
  - ✅ `test_import_backup_rejects_zip_without_database` - Database file requirement
  - ✅ `test_import_backup_handles_multiple_database_locations` - Multiple DB path support
  - ✅ `test_import_backup_fails_in_demo_mode` - Demo mode restriction
  - ✅ `test_import_backup_validates_mime_types` - Enhanced MIME type validation

  **Coverage includes:**
  - Enhanced backup file validation (commits c8a4bb2c, 4b153f69)
  - Flexible file structure handling with multiple source paths (1cd0a549)
  - File::exists usage for path validation (b7ed30d8)
  - All the refactored backup import logic

erhanurgun committed:
add tests for backup import functionality and validation

saeedvaziry committed:
Merge branch '3.x' into fix/backup-import-storage-path

saeedvaziry commented:
<img width="3638" height="1736" alt="image" src="https://github.com/user-attachments/assets/a6521446-f2de-4f1c-be9d-6ad735de794e" />

saeedvaziry commented:
Use `move_directory` helper instead

erhanurgun committed:
replace File::move with move_directory for improved backup import handling

erhanurgun committed:
Resolved Issues:

  1. Test database issue: The storage/database-test.sqlite file was missing and has been created.
  2. move_directory function issue: This function was designed only for directories, but was also being used for files. Separate logic was added for files and directories in the VitoSettingController::moveBackupFiles() method:
  - For files: File::copy()
    - For directories: move_directory()
  3. Route conflict: In the PluginController.php file, the same route name (plugins.install) was used for two different methods. The route name on line 93 was changed to plugins.install.patch. (I had to fix it because it was affecting my test's performance!)

erhanurgun commented:
> Use `move_directory` helper instead

I made the change you requested... For details, see this commit message.: https://github.com/erhanurgun/FORK-laravel-vitodeploy/commit/334b6af44272516b432821a5b258df384065497c

saeedvaziry commented:
tip: run `composer pre-commit` before each commit

erhanurgun committed:
The forgotten pre-commit command was executed

saeedvaziry commented on .gitignore:
```diff
@@ -8,7 +8,7 @@
 /vendor
 /storage/database.sqlite
-/storage/database-test.sqlite
+#/storage/database-test.sqlite
```
we shouln't add test db to the git

erhanurgun replied:
I forgot to remove the # sign after my test!

saeedvaziry commented on app/Http/Controllers/PluginController.php:
```diff
@@ -90,7 +90,7 @@ public function installNewPlugin(Request $request, InstallGithubPlugin $action):
     }
 
-    #[Patch('/install', name: 'plugins.install')]
+    #[Patch('/install', name: 'plugins.install.patch')]
```
why this change?

erhanurgun replied:
During my testing process... This was causing me to get an error saying that the route name was already in use!

<img width="1753" height="421" alt="ss1" src="https://github.com/user-attachments/assets/7925b17e-10dd-47d7-bde5-fcb765bd3009" />

erhanurgun replied:
During my testing process... This was causing me to get an error saying that the route name was already in use!

<img width="1742" height="414" alt="ss1" src="https://github.com/user-attachments/assets/e8ccd08f-5028-4ee3-b73e-73411077d7e7" />

erhanurgun replied:
@saeedvaziry Do you want me to revert it back to its original state?

saeedvaziry replied:
Yes, it should not have any connection to imports

erhanurgun committed:
unignore database-test.sqlite for backup import testing

erhanurgun committed:
Merge branch 'vitodeploy:3.x' into fix/backup-import-storage-path

saeedvaziry commented:
also please don't commit package.lock

erhanurgun committed:
The package-lock file that was sent by mistake has been removed.

erhanurgun committed:
Merge branch 'fix/backup-import-storage-path' of https://github.com/erhanurgun/FORK-laravel-vitodeploy into fix/backup-import-storage-path

saeedvaziry committed:
revert package-lock

saeedvaziry committed:
revert plugin controller

saeedvaziry committed:
Merge branch '3.x' into fix/backup-import-storage-path

saeedvaziry commented:
@erhanurgun why the tests are duplicated?

erhanurgun commented:
> @erhanurgun why the tests are duplicated?

First, I wrote the tests in VitoSettingsTest.php, but unfortunately, the test results took too long and I couldn't get any output! It was because of parent::setUp()...

So I wrote VitoSettingsImportTest.php and was able to verify the accuracy of the tests.

erhanurgun commented:
@saeedvaziry Would you like me to remove the tests in VitoSettingsTest.php?

saeedvaziry commented:
Yes

erhanurgun committed:
Merge branch 'vitodeploy:3.x' into fix/backup-import-storage-path

erhanurgun committed:
refactor VitoSettingsTest by removing unused setup and import backup tests

erhanurgun committed:
The forgotten “composer pre-commit” command was executed.

saeedvaziry commented:
@erhanurgun thanks for the effort but I am going to close this PR.

running tests locally breaks my local setup completely. it replaces the env file of my local with something else (almost empty) which causes a full loss on my local setup. I was hoping to have this PR at least for the sake of the tests but even tests are not standard enough to keep.

<img width="3758" height="1844" alt="image" src="https://github.com/user-attachments/assets/04c5e9f4-acf5-4bfe-9696-46e661b28b9e" />

erhanurgun commented:
@saeedvaziry No problem... Maybe we'll try again when we switch to pest tests... PHPUnit is a bit tricky for me... It was fun for me, I love Vito :)
